### ISSUE TITLE
Blank banner on iOS

### ISSUE THREAD

**Issue by sinneren** [ain/smartbanner.js#66]
I only have on real iOS devices 9+ blank banner. Without picture too. But on devTools emulator is all OK. Because margin-top:84px on html tag. Only in safari.

**Comment by ain** [comment]
Could you post your `smartbanner` meta tags please?

**Comment by dvejmz** [comment]
Same issue here. The library shows a blank smartbanner at the top of the page, like the picture above. This only happens on iPhone devices and Safari. iPhone + Chrome is fine.

# Environment
**OS:** iOS 11
**Device:** iPhone 7

This is the user agent of the iPhone device where I found the problem

```
Mozilla/5.0 (iPhone; CPU iPhone OS 11_1_2 like Mac OS X) AppleWebKit/604.3.5 (KHTML, like Gecko) Version/11.0 Mobile/15B202 Safari/604.1
```

# Steps to reproduce
1. Install smartbanner.js on your web server (in my case it's nginx + PHP-FPM)
2. Add these meta tags to your `<head>`

```
<meta name="smartbanner:title" content="App">
<meta name="smartbanner:author" content="Me">
<meta name="smartbanner:price" content="FREE">
<meta name="smartbanner:price-suffix-google" content=" - Play Store">
<meta name="smartbanner:icon-google" content="some-icon.jpg">
<meta name="smartbanner:button" content="VIEW">
<meta name="smartbanner:button-url-google" content="https://play.google.com/store/apps/details?id=app">
<meta name="smartbanner:custom-design-modifier" content="container">
<meta name="smartbanner:enabled-platforms" content="none">
<meta name="smartbanner:hide-path" content="/">
```
3. Navigate to your website from an iPhone using Safari.

# What happens

A blank banner is shown at the top of the page. See picture below

![img_fbadfaf60e9c-1](https://user-images.githubusercontent.com/9487006/33386763-a7d2be4e-d523-11e7-867a-c0f315e3f7a4.jpeg)

As I said, this only happens if you navigate to the page using an iPhone device on Safari. Using Chrome on an iPhone doesn't show a blank banner. Android works fine too.

# What should happen

The blank banner does not appear.

**Comment by ain** [comment]
Please follow [Basic usage](https://github.com/ain/smartbanner.js#basic-usage) for default options. 

In the above example you've shared almost all Apple metadata is missing, e.g. `smartbanner:icon-apple`, `smartbanner:price-suffix-apple`, `smartbanner:button-url-apple`.

**Comment by dvejmz** [comment]
Thanks for your reply.

Yes, that is intentional. In my case, I only wish to display the smartbanner for the Android app and show no banner at all in iOS devices. Since I don't want a smartbanner for iOS at all I figured I wouldn't need any metadata for it. Do I need Apple metadata even if I don't want to show the Apple smartbanner at all?

**Comment by ain** [comment]
@dvejmz in that case, did you try setting `smartbanner:enabled-platforms` to `android`? 

User Agent filtering we have is quite advanced, please see [Show the smartbanner for certain User Agents](https://github.com/ain/smartbanner.js#show-the-smartbanner-for-certain-user-agents) for more information.

**Comment by dvejmz** [comment]
@ain I just got the bottom of the problem, and it was entirely not an issue with smartbanner.js! As you said, the user agent detection of your library works absolutely fine with iOS 11 + Safari. The problem was that in my page I had this Apple app store meta tag alongside smartbanner.js

```
<meta name="apple-itunes-app" content="app-id=<id>">
```

However, since I was testing this on my local dev environment which is not properly configured to show actual Apple smart banners like that, that blank smartbanner was showing instead. So yeah, as I said, nothing to do with your library. To verify this I even checked out smartbanner.js, added another unit test with the exact same user agent of the iPhone I was having the issue on, ran the tests, all passing, then started the local smartbanner.js server and messed around with the source a bit, nothing, all was working just fine!

I can't say for sure but I'd say this is the same problem @sinneren was having so given I've verified myself that this library is indeed not the problem and that the OP hasn't really got back to you on this, I reckon it's safe to close this one off for good!

Thank you very much for your help.
